src/context/HistoryContext.jsx

```diff
@@ -0,0 +1,98 @@
+import {useContext, useReducer, createContext, useEffect} from 'react';
+import { useAuth } from './AuthContext';
+import {HistoryReducer} from "../reducer/HistoryReducer";
+import axios from 'axios';
+
+
+const HistoryContext = createContext();
+
+const HistoryProvider = ({children}) => {
+    const {token} = useAuth()
+    const [historyState, historyDispatch] = useReducer(HistoryReducer, {historyVideo: []})
+    const {historyVideo} = historyState;
+    
+    useEffect(() => {
+        if(token) {
+            (async () => {
+                try {
+                    const {data: {history}, status} = await axios.get("api/user/history", {
+                        headers: {
+                            authorization: token
```

Comment:
Suggestion: instead of settting authorization for each API call you can set a comman header :
axios.defaults.headers.common['Authorization'] = 'AUTH_TOKEN';
try this.